### PR TITLE
firefox: Added checks for new addon behaviour since v91

### PR DESCRIPTION
**Pull request by Qubasa**
###### Motivation for this change
Firefox v91 does not support addons with invalid signature anymore. Firefox ESR needs to be used for nix addon support.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix
Comment (ajs124) [inline pkgs/top-level/all-packages.nix:24473]: firefox-esr-unwrapped being 91.x while firefox-esr is 78.x seems… interesting

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 133504` at ab4f238b run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>firefox-esr-unwrapped</li>
    <li>nixos-install-tools</li>
  </ul>
</details>

File: pkgs/top-level/all-packages.nix
Comment (ajs124) [reply pkgs/top-level/all-packages.nix:24473]: No. Just look through the commit history.

**Review by mweinelt — CHANGES_REQUESTED** [review]
> I will bump firefox-esr

That's honestly not up to you, leave that to us maintainers, please.


> a mistake by a previous committer as firefox-esr-wayland is v91.

The sole reason I created `firefox-esr-wayland` based on 91esr is that I didn't want to introduce this on old 78esr, which I never actually tested on wayland. I'm much more confident in 91 for this and there didn't need to be any backward compat, because it was newly introduced.

File: doc/builders/packages/firefox.section.md
Comment (mweinelt) [inline doc/builders/packages/firefox.section.md:9]: Only in ESR builds meaning 91.0 ESR works, but 91.0 does not?

File: nixos/doc/manual/from_md/release-notes/rl-2111.section.xml
Comment (mweinelt) [inline nixos/doc/manual/from_md/release-notes/rl-2111.section.xml:298]: That seems unrelated.

File: nixos/doc/manual/release-notes/rl-2111.section.md
Comment (mweinelt) [inline nixos/doc/manual/release-notes/rl-2111.section.md:115]: Stray change.

**Review by mweinelt — CHANGES_REQUESTED** [review]
Needs more clarification about 91.0 vs 91.0 ESR, also carries a stray change that needs to be removed.

File: doc/builders/packages/firefox.section.md
Comment (Luis-Hebendanz) [reply doc/builders/packages/firefox.section.md:9]: correct

File: nixos/doc/manual/from_md/release-notes/rl-2111.section.xml
Comment (Luis-Hebendanz) [reply nixos/doc/manual/from_md/release-notes/rl-2111.section.xml:298]: Oh yeah one moment

File: doc/builders/packages/firefox.section.md
Comment (mweinelt) [reply doc/builders/packages/firefox.section.md:9]: Maybe say 78/91 ESR to be clearer here. We can likely expect 91 ESR to be the last ESR series to allow for unsigned addons.

Comment (Luis-Hebendanz) [reply doc/builders/packages/firefox.section.md]: Hmm I don't think so because there needs to be a way for corporations to deploy internal addons not published in the firefox addon store~ I rewrote it to `Nix firefox addons only work in the firefox-esr package` . Is that okay?

**Pull request merged**
